lib/src/value/color.ts

```diff
@@ -0,0 +1,373 @@
+      lightness: fuzzyAssertInRange(lightness, 0, 100, 'lightness'),
+      alpha: alpha === undefined ? 1 : fuzzyAssertInRange(alpha, 0, 1, 'alpha'),
+    });
+  }
+
+  /**
+   * Creates an HWB color.
+   *
+   * Throws an error if `whiteness` or `blackness` aren't between `0` and `100`,
+   * or if `alpha` isn't between `0` and `1`.
+   */
+  static hwb(
+    hue: number,
+    whiteness: number,
+    blackness: number,
+    alpha?: number
+  ) {
+    // From https://www.w3.org/TR/css-color-4/#hwb-to-rgb
+    const scaledHue = (hue % 360) / 360;
+    let scaledWhiteness =
+      fuzzyAssertInRange(whiteness, 0, 100, 'whiteness') / 100;
+    let scaledBlackness =
+      fuzzyAssertInRange(blackness, 0, 100, 'blackness') / 100;
+
+    const sum = scaledWhiteness + scaledBlackness;
+    if (sum > 1) {
+      scaledWhiteness /= sum;
+      scaledBlackness /= sum;
+    }
+
+    const factor = 1 - scaledWhiteness - scaledBlackness;
+    function toRgb(hue: number) {
+      const channel = hueToRgb(0, 1, hue) * factor + scaledWhiteness;
+      return fuzzyRound(channel * 255);
+    }
+
+    // Because HWB is (currently) used much less frequently than HSL or RGB, we
+    // don't cache its values because we expect the memory overhead of doing so
+    // to outweigh the cost of recalculating it on access. Instead, we eagerly
+    // convert it to RGB and then convert back if necessary.
+    return SassColor.rgb(
+      toRgb(scaledHue + 1 / 3),
+      toRgb(scaledHue),
+      toRgb(scaledHue - 1 / 3),
+      alpha
+    );
+  }
+
+  /** `this`'s red channel. */
+  get red(): number {
+    if (this.redInternal === undefined) {
+      this.hslToRgb();
+    }
+    return this.redInternal!;
+  }
+
+  /** `this`'s blue channel. */
+  get blue(): number {
+    if (this.blueInternal === undefined) {
+      this.hslToRgb();
+    }
+    return this.blueInternal!;
+  }
+
+  /** `this`'s green channel. */
+  get green(): number {
+    if (this.greenInternal === undefined) {
+      this.hslToRgb();
+    }
+    return this.greenInternal!;
+  }
+
+  /** `this`'s hue value. */
+  get hue(): number {
+    if (this.hueInternal === undefined) {
+      this.rgbToHsl();
+    }
+    return this.hueInternal!;
+  }
+
+  /** `this`'s saturation value. */
+  get saturation(): number {
+    if (this.saturationInternal === undefined) {
+      this.rgbToHsl();
+    }
+    return this.saturationInternal!;
+  }
+
+  /** `this`'s hue value. */
+  get lightness(): number {
+    if (this.lightnessInternal === undefined) {
+      this.rgbToHsl();
+    }
+    return this.lightnessInternal!;
+  }
+
+  /** `this`'s whiteness value. */
+  get whiteness(): number {
+    // Because HWB is (currently) used much less frequently than HSL or RGB, we
+    // don't cache its values because we expect the memory overhead of doing so
+    // to outweigh the cost of recalculating it on access.
+    return (Math.min(Math.min(this.red, this.green), this.blue) / 255) * 100;
```

Comment:
```suggestion
    return Math.min(this.red, this.green, this.blue) / 255 * 100;
```

`Math.min()` takes any number of arguments. Same with `Math.max()` below.